### PR TITLE
Save indexes to database and retrieve them

### PR DESCRIPTION
**Pull request by devjoaov**

### CODE REVIEW

File: pkg/op-brains/op_brains/documents/__init__.py [hunk 1]
```diff
@@ -9,9 +9,9 @@
 import time
 
 chat_sources = [
-    [
-        FragmentsProcessingStrategy,
-    ],
+    # [
+    #     FragmentsProcessingStrategy,
+    # ],
```
Comment (ribeirojose) [inline pkg/op-brains/op_brains/documents/__init__.py:14]: how come? 🤔

File: pkg/op-data/op_data/db/models.py [hunk 2]
```diff
@@ -61,12 +61,22 @@ class RawTopic(Model):
...
         table = "RawTopic"
 
+class Embedding(Model):
+    id = fields.IntField(pk=True)
+    compressedData = fields.BinaryField(required=True)
+    createdAt = NaiveDatetimeField(auto_now_add=True)
+    updatedAt = NaiveDatetimeField(auto_now=True)
+
+    class Meta:
+        table = "Embeddings"
```
Comment (ribeirojose) [inline pkg/op-data/op_data/db/models.py:78]: don't we need an embedding type? I think we generate multiple embeddings for each context right?

File: pkg/op-brains/op_brains/documents/__init__.py (hunk 1 above)
Comment (devjoaov) [reply pkg/op-brains/op_brains/documents/__init__.py:14]: I sent a message on slack about this https://bleu-builders.slack.com/archives/C0740EA570D/p1725553560473179

File: pkg/op-data/op_data/db/models.py (hunk 2 above)
Comment (devjoaov) [reply pkg/op-data/op_data/db/models.py:78]: the compressed data is a json where the keys are the old `db_name`. in short, is a folder with the same keys we used on the `dbs` folder
image

File: pkg/op-brains/op_brains/chat/test_chat_model.py [hunk 3]
```diff
@@ -19,6 +19,20 @@ def invoke(self, prompt, max_tokens=10000):
         )
         return self.Response(response.choices[0].message.content)
 
+    async def ainvoke(self, prompt, max_tokens=2000):
```
Comment (ribeirojose) [inline pkg/op-brains/op_brains/chat/test_chat_model.py:22]: would we keep this or use mixtral OR even openai?

File: pkg/op-brains/op_brains/chat/test_chat_model.py [hunk 4]
```diff
@@ -1,12 +1,12 @@
-
 import groq
+import asyncio
 
 class GroqLLM:
-    def __init__(self, model_name='mixtral-8x7b-32768'):
+    def __init__(self, model_name='llama3-8b-8192'):
```
Comment (ribeirojose) [inline pkg/op-brains/op_brains/chat/test_chat_model.py:5]: would we keep this or use mixtral OR even openai?

**Review by ribeirojose — APPROVED** [review]
🧙

File: pkg/op-brains/op_brains/chat/test_chat_model.py (hunk 4 above)
Comment (devjoaov) [reply pkg/op-brains/op_brains/chat/test_chat_model.py:5]: this class is just for testing purpose

**Pull request merged**